framework/src/main/java/org/checkerframework/framework/type/ViewpointAdapter.java

```diff
@@ -0,0 +1,65 @@
+package org.checkerframework.framework.type;
+
+import java.util.List;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+
+/**
+ * Interface that contains viewpoint adaptation logic.
+ *
+ * <p>Standard viewpoint adaptation happens on members/fields accesses, constructor invocations,
+ * method invocations and type parameter instantiations.
+ */
+public interface ViewpointAdapter {
+
+    /**
+     * Viewpoint adapts members/fields accesses
+     *
+     * @param receiverType receiver type through which members are accessed
+     * @param memberElement element of the accessed member
+     * @param memberType accessed type of the member. After the method returns, it will be mutated
+     *     to the viewpoint adapted result.
+     */
+    void viewpointAdaptMember(
+            AnnotatedTypeMirror receiverType,
+            Element memberElement,
+            AnnotatedTypeMirror memberType);
```

Comment:
Type arguments are correctly substituted, and memberType is already in a good shape. Only annotations should be replaced by the viewpoint adapted ones.